source/data/texture/TextureLoader.cpp

```diff
@@ -31,45 +32,59 @@ Texture TextureLoader::getEmptyNormal() {
 }
 
 Texture TextureLoader::getEmptySpecular() {
-  return TextureLoader::getEmptyTexture("engine@empty/specular", "\x7F\x7F\x7F");
+  return TextureLoader::getEmptyTexture("engine@empty/specular",
+                                        "\x7F\x7F\x7F");
 }
 
 Texture TextureLoader::getTexture(const std::string &path) {
+  // Check if texture is chached
   auto it = textureCache.find(path);
-  if (it != textureCache.end()) {
+  if (it != textureCache.end())
     return it->second;
-  }
-  int width = 0, height = 0;
-  FIBITMAP *bitmap = FreeImage_Load(FreeImage_GetFileType((Environment::getDataDir() + "/textures/" + path).c_str()), (Environment::getDataDir() + "/textures/" + path).c_str());
+
+  // Read image from hard
+  FIBITMAP *bitmap = FreeImage_Load(
+      FreeImage_GetFileType(
+          (Environment::getDataDir() + "/textures/" + path).c_str()),
+      (Environment::getDataDir() + "/textures/" + path).c_str());
```

Comment:
Actually these two lines (even before your commit) are incorrect: the generated `std::string` is temporary and *may* die before invocation is finished since `FreeImage_GetFileType` and `FreeImage_Load` expects `const char*` parameters, and not an `std::string` rvalue reference. By the time the `.c_str()` has finished evaluating, the strings are destructed, and the pointed-to memory is potentially invalid.

Consider moving the string concat to a variable while you're at it.